src/fiber/graph.rs

```diff
@@ -828,6 +825,33 @@ where
             ));
         }
 
+        let hop_hint_map: HashMap<Pubkey, (OutPoint, bool)> = hop_hints
```

Comment:
I think the map type should be `HashMap<(Pubkey, bool), OutPoint>`, otherwise this case will go wrong:
```
we want to rebalance the channels 1 and 4, and provide two hints (c1/c4) for the payment
A -- Channel 1 --> B
A -- Channel 2 --> B
B -- Channel 3 --> A
B -- Channel 4 --> A
``` 

could you add an unit test for this case?

Reply:
fixed with commit https://github.com/nervosnetwork/fiber/pull/432/commits/d8c824bbb91615bb1b1ae397054b99734fb1c8fa